Road/source/features/data_cleaning.py

```diff
@@ -0,0 +1,193 @@
+    """
+    url_pattern = re.compile(r'https?://\S+|www\.\S+')
+    return url_pattern.sub(r'', text)
+
+def remove_phone_numbers(text: str) -> str:
+    """
+    Remove phone numbers from text.
+
+    Args:
+        text (str): The input text with phone numbers.
+
+    Returns:
+        str: The input text with phone numbers removed.
+    """
+    text = re.sub(r'(\+?\d{2,4}[ -]?)?\d{9,10}', '', text)
+    return text
+
+def remove_punctuations(text: str) -> str:
+     """
+     Remove punctuations from text.
+
+     Args:
+          text (str): The input text with punctuations.
+
+     Returns:
+          str: The input text with punctuations removed.
+     """
+     return text.translate(str.maketrans('', '', PUNCT_TO_REMOVE))
+
+def remove_multimedia(text: str) -> str:
+    """
+    Remove multimedia (images, videos, etc.) from text.
+
+    Args:
+        text (str): The input text with multimedia.
+
+    Returns:
+        str: The input text with multimedia removed.
+    """
+    text = re.sub(r'http\S+', '', text)
+    text = re.sub(r'pic\.twitter\.com/\S+', '', text)
+    text = re.sub(r'@(\w+)', '', text)
+    return text
+
+def map_reply_messages(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Map each message to its corresponding reply message in the "reply" column, and delete any rows with non -1 reply_to values.
+
+    Args:
+        data (pandas.DataFrame): The input DataFrame containing message data.
+
+    Returns:
+        pandas.DataFrame: The modified DataFrame with mapped reply messages and deleted rows.
+    """
+    # create an empty dictionary to store message-reply mappings
+    message_reply_dict = {}
+
+    # iterate over each row in the dataframe
+    for i, row in data.iterrows():
+        # check if the row has a valid reply_to value
+        if row["reply_to"] != -1:
+            # store the message-reply mapping in the dictionary
+            message_reply_dict[row["reply_to"]] = row["message"]
+
+    # update the reply column with the corresponding reply message
+    data["reply"] = data["id"].map(message_reply_dict)
+
+
+    return data
+
+
+def clean_text(text: str) -> str:
+    """
+    This function takes a string input and performs various cleaning operations.
+    The cleaned text is returned as output.
+
+    Args:
+        text (str): The input string to be cleaned.
+
+    Returns:
+        str : The cleaned text string.
+    """
+    
+    # Remove diacritics (accents) from text
+    text = remove_diacritics(text)
+    
+    # Remove emojis and emoticons from text
+    text = remove_emojis_emoticons(text)
+    
+    # Remove URLs from text
+    text = remove_urls(text)
+    
+    # Remove phone numbers from text
+    text = remove_phone_numbers(text)
+    
+    # Remove punctuation marks from text
+    text = remove_punctuations(text)
+    
+    # Remove multimedia elements from text
+    text = remove_multimedia(text)
+
+    # Return the cleaned text
+    return text
```

Comment:
@eleenkmail This code should not be here, it must be at the `endpoint.py` file

Reply:
Not really, in endpoint, there is a function which calls this function to facilitate applying all these steps on a string in one line